WooCommerce/Classes/Extensions/UIAlertController+Helpers.swift

```diff
@@ -29,6 +29,35 @@ extension UIAlertController {
         viewController.present(actionSheet, animated: true)
     }
 
+    /// Present a discard changes action sheet when the user navigates away from adding a product.
+    static func presentDiscardNewProductActionSheet(viewController: UIViewController,
+                                                    onSaveDraft: @escaping (() -> Void),
+                                                    onDiscard: @escaping (() -> Void),
+                                                    onCancel: (() -> Void)? = nil) {
+        let actionSheet = UIAlertController(title: nil, message: ActionSheetStrings.message, preferredStyle: .actionSheet)
+        actionSheet.view.tintColor = .text
+
+        actionSheet.addDefaultActionWithTitle(ActionSheetStrings.saveAsDraft) { _ in
+            onSaveDraft()
+        }
+
+        actionSheet.addDestructiveActionWithTitle(ActionSheetStrings.discard) { _ in
+            onDiscard()
+        }
+
+        actionSheet.addCancelActionWithTitle(ActionSheetStrings.cancel) { _ in
+            onCancel?()
+        }
+
+        if let popoverController = actionSheet.popoverPresentationController {
+            popoverController.sourceView = viewController.view
+            popoverController.sourceRect = viewController.view.bounds
+            popoverController.permittedArrowDirections = []
```

Comment:
[nit] Should we pass the source button so this looks better positioned on iPad?

Reply:
Since the discard changes action sheet is triggered by tapping on the back navigation bar button or swipe-back gesture, we currently don't anchor the bottom sheet at a particular button and just show the popover in the middle of the view:

![Simulator Screen Shot - iPad Air 2 - 2020-09-24 at 10 50 02](https://user-images.githubusercontent.com/1945542/94096693-4915d980-fe57-11ea-86ba-c04cf0260a42.png)

I tried passing `navigationItem.backBarButtonItem` to see if we can anchor the popover at the back button if it's triggered by the button tap action. However, I think `navigationItem.backBarButtonItem` is to be displayed on the next view controller and not the current one (the app crashes because it's not anchoring at a valid view).